snap/snapcraft.yaml

```diff
@@ -106,21 +77,22 @@ parts:
       install -DT "./example/cmd/device-simple/device-simple" \
         "$SNAPCRAFT_PART_INSTALL/bin/device-simple"
       
-      install -d "$SNAPCRAFT_PART_INSTALL/config/device-simple/res/"
-
       # "ProfilesDir" in combination with the confdir and profile command-line
       # options are sufficient to not need to cd before exec'ing the binary.
       # Change "ProfilesDir" to be under $SNAP_DATA which is writable and 
       # where the install hook copies the config files and device profiles to
       # when the snap is first installed 
       cat "./example/cmd/device-simple/res/configuration.toml" | \
         sed -e s:\"./device-simple.log\":\'\$SNAP_COMMON/device-simple.log\': \
-          -e s:'ProfilesDir = \"./res\"':'ProfilesDir = \"\$SNAP_DATA/config/device-simple/res\"': > \
+        -e s:"ProfilesDir = './res/profiles'":"ProfilesDir = '\$SNAP_DATA/config/device-simple/res/profiles'":  \
+        -e s:"DevicesDir = './res/devices'":"DevicesDir = '\$SNAP_DATA/config/device-simple/res/devices'": > \
```

Comment:
This `sed` statement can just be replaced by a cp command, and then these both can just be overridden with environment variables in the app stanza above. Ex.

    command: bin/device-simple $CONFIG_PRO_ARG $CONF_ARG $REGISTRY_ARG
    # Since we can't include the colon ':' characters in the command above,
    # we need to specify the consul address in an env var like so
    # see https://bugs.launchpad.net/snapd/+bug/1827392 for more details
    environment:
      REGISTRY_ARG: "--registry"
      CONFIG_PRO_ARG: "--cp=consul://localhost:8500"
      CONF_ARG: "--confdir=$SNAP_DATA/config/device-simple/res"
      DEVICE_PROFILESDIR: "$SNAP_DATA/config/device-simple/res/profiles"
      DEVICE_DEVICESDIR: "$SNAP_DATA/config/device-simple/res/devices"